File: biblioteka.cs
using System;
using System.Collections.Generic;

class Ksiazka
{
    public int id_k;
    public string tytul;
    public string autor;
    public bool dostepnosc_k = true;
}

class Uzytkownik
{
    public string imie;
    public string nazwisko;
    public int id_u;
}

class Wypozyczania
{
    public Ksiazka w_ksiazka;
    public Uzytkownik w_uzytkownik;
}

class Biblioteka
{
    public List<Ksiazka> ksiazki = new List<Ksiazka>();
    public List<Uzytkownik> uzytkownicy = new List<Uzytkownik>();
    public List<Wypozyczania> wypozyczenia = new List<Wypozyczania>();
    private int nastepneID_uzytkownika = 1;

    public void dodaj_ksiazke(int id_k, string tytul, string autor)
    {
        Ksiazka ksiazka = new Ksiazka() { id_k = id_k, tytul = tytul, autor = autor };
        ksiazki.Add(ksiazka);
    }

    public int dodaj_uzytkownika(string imie, string nazwisko)
    {
        Uzytkownik uzytkownik = new Uzytkownik()
        {
            imie = imie,
            nazwisko = nazwisko,
            id_u = nastepneID_uzytkownika++
        };
        uzytkownicy.Add(uzytkownik);
        Console.WriteLine($"Zostales dodany do systemu twoje imie:{uzytkownik.imie} twoje nazwisko: {uzytkownik.nazwisko} twoje ID: {uzytkownik.id_u}");
        return uzytkownik.id_u;
    }

    public void wypozycz_ksiazke(int id_ksiazki, int id_u)
    {
        Ksiazka ksiazka = ksiazki.Find(k => k.id_k == id_ksiazki && k.dostepnosc_k);
        Uzytkownik uzytkownik = uzytkownicy.Find(u => u.id_u == id_u);

        if (ksiazka != null && uzytkownik != null)
        {
            Wypozyczania nowe = new Wypozyczania()
            {
                w_ksiazka = ksiazka,
                w_uzytkownik = uzytkownik
            };
            wypozyczenia.Add(nowe);
            ksiazka.dostepnosc_k = false;

            Console.WriteLine($"wybrales tytul {ksiazka.tytul} twoje imie:{uzytkownik.imie} twoje nazwisko; {uzytkownik.nazwisko}.");
        }
        else
        {
            Console.WriteLine("Brak ksiazki w systemie");
        }
    }

    public void pokaz_dostepne_ksiazki()
    {
        Console.WriteLine("\n Ksiazki znajdujace sie w bibliotece:");
        foreach (var ksiazka in ksiazki)
        {
            if (ksiazka.dostepnosc_k)
                Console.WriteLine($"ID: {ksiazka.id_k} -{ksiazka.tytul} ----{ksiazka.autor}");
        }
    }
}

class Program
{
    static void Main()
    {
        Biblioteka moja_biblioteka = new Biblioteka();

        moja_biblioteka.dodaj_ksiazke(1, "Krzyzacy", "Henryk Sienkiewicz");
        moja_biblioteka.dodaj_ksiazke(2, "Lalka", "Bolesław Prus");
        moja_biblioteka.dodaj_ksiazke(3, "Pan Tadeusz", "Adam Mickiewicz");
        moja_biblioteka.dodaj_ksiazke(4,"W pustyni i w puszczy", "Henryk Sienkiewicz");
        moja_biblioteka.dodaj_ksiazke(5,"Zbrodnia i kara", "Fiodor Dostojewski");
        moja_biblioteka.dodaj_ksiazke(6,"1984", "George Orwell");
        moja_biblioteka.dodaj_ksiazke(7,"Dziady", "Adam Mickiewicz");

        Console.Write("Podaj swoje imie: ");
        string imie = Console.ReadLine();

        Console.Write("Podaj swoje nazwisko: ");
        string nazwisko = Console.ReadLine();

        int id_u = moja_biblioteka.dodaj_uzytkownika(imie, nazwisko);

        moja_biblioteka.pokaz_dostepne_ksiazki();

        Console.Write("Wybierz ksiazke - podaj ID: ");
        int id_ksiazki = int.Parse(Console.ReadLine());

        moja_biblioteka.wypozycz_ksiazke(id_ksiazki, id_u);
    }
}

File: biblioteka.py
class Ksiazka:
    def d_ksiazki(self, id_k, tytul, autor):
        self.id_k = id_k
        self.tytul = tytul
        self.autor = autor
        self.dostepnosc_k = True

class Uzytkownik:
    def d_uzytkownik(self, imie, nazwisko, id_u):
        self.imie = imie
        self.nazwisko = nazwisko
        self.id_u = id_u

class Wypozyczanie:
    def d_wypozyczenie(self, ksiazka, uzytkownik):
        self.w_ksiazka = ksiazka
        self.w_uzytkownik = uzytkownik

class Biblioteka:
    def d_B(self):
        self.ksiazki = []
        self.uzytkownicy = []
        self.wypozyczenia = []
        self.nastepne_id_uzytkownika = 1

    def dodaj_ksiazke(self, id_k, tytul, autor):
        ksiazka = Ksiazka()
        ksiazka.d_ksiazki(id_k, tytul, autor)
        self.ksiazki.append(ksiazka)

    def dodaj_uzytkownika(self, imie, nazwisko):
        uzytkownik = Uzytkownik()
        uzytkownik.d_uzytkownik(imie, nazwisko, self.nastepne_id_uzytkownika)
        self.nastepne_id_uzytkownika += 1
        self.uzytkownicy.append(uzytkownik)
        print(f"Dodano do systemu-Imię: {uzytkownik.imie}, -Nazwisko: {uzytkownik.nazwisko}, -ID: {uzytkownik.id_u}")
        return uzytkownik.id_u

    def wypozycz_ksiazke(self, id_ksiazki, id_u):
        ksiazka = next((k for k in self.ksiazki if k.id_k == id_ksiazki and k.dostepnosc_k), None)
        uzytkownik = next((u for u in self.uzytkownicy if u.id_u == id_u), None)

        if ksiazka and uzytkownik:
            wypozyczenie = Wypozyczanie()
            wypozyczenie.d_wypozyczenie(ksiazka, uzytkownik)
            self.wypozyczenia.append(wypozyczenie)
            ksiazka.dostepnosc_k = False
            print(f"posiadana książka: {ksiazka.tytul}, Imię: {uzytkownik.imie}, Nazwisko: {uzytkownik.nazwisko}.")
        

    def zwroc_ksiazke(self, id_ksiazki, id_u):
        wypozyczenie = next((w for w in self.wypozyczenia if w.w_ksiazka.id_k == id_ksiazki and w.w_uzytkownik.id_u == id_u), None)
        if wypozyczenie:
            wypozyczenie.w_ksiazka.dostepnosc_k = True
            self.wypozyczenia.remove(wypozyczenie)
            print(f"książka zostala zwrócona")
        else:
            print("Podałes złe ID książki.")

    def pokaz_dostepne_ksiazki(self):
        print("\nKsiążki dostępne w bibliotece:")
        for ksiazka in self.ksiazki:
            if ksiazka.dostepnosc_k:
                print(f"ID: {ksiazka.id_k} - {ksiazka.tytul} --- {ksiazka.autor}")

biblioteka = Biblioteka()
biblioteka.d_B()

biblioteka.dodaj_ksiazke(1, "Krzyżacy", "Henryk Sienkiewicz")
biblioteka.dodaj_ksiazke(2, "Lalka", "Bolesław Prus")
biblioteka.dodaj_ksiazke(3, "Pan Tadeusz", "Adam Mickiewicz")
biblioteka.dodaj_ksiazke(4, "W pustyni i w puszczy", "Henryk Sienkiewicz")
biblioteka.dodaj_ksiazke(5, "Zbrodnia i kara", "Fiodor Dostojewski")
biblioteka.dodaj_ksiazke(6, "1984", "George Orwell")
biblioteka.dodaj_ksiazke(7, "Dziady", "Adam Mickiewicz")

imie = input("Podaj swoje imię: ")
nazwisko = input("Podaj swoje nazwisko: ")
id_u = biblioteka.dodaj_uzytkownika(imie, nazwisko)

biblioteka.pokaz_dostepne_ksiazki()

id_ksiazki = int(input("Wybierz książkę-podaj ID: "))
biblioteka.wypozycz_ksiazke(id_ksiazki, id_u)
id_ksiazki_zwrot = int(input("Wybierz książkę którą chcesz zwrocic-podaj ID: "))
biblioteka.zwroc_ksiazke(id_ksiazki_zwrot, id_u)
